src/status_im2/contexts/onboarding/common/carousel/view.cljs

```diff
@@ -54,45 +40,48 @@
    [rn/view {:style (style/progress-bar-item static? true)}]])
 
 (defn dynamic-progress-bar
-  [progress-bar-width animate?]
+  [progress-bar-width animate? progress]
   [:f>
    (fn []
-     (let [width          (animation/dynamic-progress-bar-width progress-bar-width animate?)
+     (let [width          (animation/dynamic-progress-bar-width progress-bar-width animate? progress)
            container-view (if animate? reanimated/view rn/view)]
        [container-view {:style (style/dynamic-progress-bar width animate?)}
         [progress-bar
          {:static?            false
           :progress-bar-width progress-bar-width}]]))])
 
 (defn view
-  [animate?]
+  [{:keys [animate? progress paused header-text content paused? header-background]}]
   [:f>
    (fn []
      (let [window-width       (rf/sub [:dimensions/window-width])
            view-id            (rf/sub [:view-id])
            status-bar-height  (:status-bar-height @navigation/constants)
            progress-bar-width (- window-width 40)
-           carousel-left      (animation/carousel-left-position window-width animate?)
+           carousel-left      (animation/carousel-left-position window-width animate? progress)
            container-view     (if animate? reanimated/view rn/view)]
-       (when animate?
-         (rn/use-effect
-          (fn []
-            (reanimated/set-shared-value @animation/paused (not= view-id :intro)))
-          [view-id]))
+       (rn/use-effect
+        (fn []
+          (when animate?
```

Comment:
yes, previously it had 
```
(when animate?
(rn/use-effect
        (fn [] ... )))
          ```
 but this looks dangerous to me as hooks should not be conditionally called. React will break if one render has a different number of hooks called then the previous render and so other issues like this.
 
 So I thought this was a safer option

Reply:
> hooks should not be conditionally called

100%

My comment was more related to the fact that if `animate?` changes, the hook won't rerun and it can be a source of a subtle bug. If `animate?` becomes false, the `paused` state won't be re-set. But I was just double-checking if it was your intention or not. Thanks!

Reply:
ok gotcha, yes in this case it was intentional. I wanted the hook to retrigger on view-id changing because it's supposed to pause/ unpause whenever it gets to the right page. I adjusted it slightly in my latest commit but it's similar